packages/react-storage/src/components/StorageBrowser/Views/Controls/Navigate.tsx

```diff
@@ -93,23 +91,46 @@ export interface NavigateControl<
 }
 
 export const NavigateControl: NavigateControl = (_props) => {
+  const [state, handleUpdateState] = useControl({ type: 'NAVIGATE' });
+
   return (
     <NavigateContainer>
       <NavigateItem
-        item={{
-          name: 'Home',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-      />
-      <NavigateItem
-        item={{
-          name: 'Location01',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-        current
+        item={HOME_NAVIGATE_ITEM}
+        current={!state.location}
+        onClick={() => handleUpdateState({ type: 'DESELECT_LOCATION' })}
       />
+      {state.location && (
+        <>
+          <NavigateItem
+            item={state.location.bucket}
+            current={!state.history || state.history.length === 0}
+            onClick={() => {
+              if (state?.location) {
+                handleUpdateState({
+                  type: 'SELECT_LOCATION',
+                  location: state.location,
+                });
+              }
+            }}
+          />
+          {state.history?.map((folder, index) => (
+            <NavigateItem
+              key={index}
+              item={folder}
+              current={folder === state.history?.[state.history.length - 1]}
+              onClick={() => {
+                if (isFolderName(folder)) {
```

Comment:
Same question as https://github.com/aws-amplify/amplify-ui/pull/5494/files#r1700996963 for this conditional

Reply:
When trying to pass `folder` to `handleUpdateState`, typescript complains that `Type 'string' is not assignable to type '`${string}/`'.ts(2322)` since `name` is of type `FolderName`

Added this type guard so that it doesn't complain

Reply:
Okay cool, it might make sense to remove that type but we can worry abt that later 😅